File: front/src/app/models/ClaseChat.ts
import { MensajeChat } from './MensajeChat';

export class ClaseChat {
	public idClase: number;

	public idCurso: number;

	public nombreClase: string;

	public mensajes: MensajeChat[];

	constructor(_idClase: number, _idCurso: number, _nombreClase: string, _mensajes: MensajeChat[]) {
		this.idClase = _idClase;
		this.idCurso = _idCurso;
		this.nombreClase = _nombreClase;
		this.mensajes = _mensajes;
	}
}

File: sov_school_back/back_chat/src/main/java/com/sovereingschool/back_chat/Models/ClaseChat.java
package com.sovereingschool.back_chat.Models;

import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ClaseChat {

    /** Clase ID */
    private Long idClase;

    /** Curso ID */
    private Long idCurso;

    /** Lista de ids de documentos de mensajes */
    private List<String> mensajes;

}